### ISSUE TITLE
expected_token gives wrong location for typo in generic parameter declaration

### ISSUE THREAD

**Issue by Hixie** [dart-lang/sdk#38732]
```dart
class A<T as Object> { }
```
->
```
  error • Expected to find '>' • test.dart:1:9 • expected_token
```

That's just after the `>`. This is misleading. The actual error is that `as` is the wrong keyword, but the error makes it look like the problem is that the `<` isn't valid or something.

**Comment by bwilkerson** [comment]
@jensjoha

**Comment by mohak852** [comment]
I'm waiting for you to assign this to me
so I can start work on it

**Comment by Hixie** [comment]
You can work on it without it being assigned to you, don't worry! :-)

**Comment by bwilkerson** [comment]
Absolutely! But please assign it to yourself when you do start working on it so that others will know that it's being worked on (or add a comment if you can't assign it to yourself).
